Add `isnull` Filter on Nullable Fields via Filter Field Factory (#4378)
nautobot/nautobot: nautobot/docs/user-guide/platform-functionality/rest-api/filtering.md

## Lookup Expressions

Certain model fields (including, in Nautobot 1.4.0 and later, custom fields of type `text`, `url`, `select`, `integer`, and `date`) also support filtering using additional lookup expressions. This allows for negation and other context-specific filtering.

These lookup expressions can be applied by adding a suffix to the desired field's name, e.g. `mac_address__n`. In this case, the filter expression is for negation and it is separated by two underscores. Below are the lookup expressions that are supported across different field types.

### Nullable Fields

> The `isnull` lookup expression is automatically added to any of the below field types, or more accurately a field type that has inherited from one of these fields (such as `DateTimeField` which inherits from `DateField`) for all core Nautobot / Nautobot App FilterSet's that inherit from `nautobot.apps.filters.BaseFilterSet`.
> 
> - `CharField`
> - `DateField`
> - `DecimalField`
> - `FloatField`
> - `IntegerField`
> - `TextField`
> - `UUIDField`
> 
> +/- 2.0.0
>     This feature was only added in Nautobot 2.0.0.
> 

Reviewer: ```suggestion
+++ 2.0.0

The `isnull` lookup expression is automatically added to any of the below field types or their subclasses, if instantiated with `null=True`, by all core Nautobot / Nautobot App FilterSet's that inherit from `nautobot.apps.filters.BaseFilterSet`.

- `CharField`
- `DateField`
- `DecimalField`
- `FloatField`
- `IntegerField`
- `TextField`
- `UUIDField`

```

## Lookup Expressions

Certain model fields (including, in Nautobot 1.4.0 and later, custom fields of type `text`, `url`, `select`, `integer`, and `date`) also support filtering using additional lookup expressions. This allows for negation and other context-specific filtering.

These lookup expressions can be applied by adding a suffix to the desired field's name, e.g. `mac_address__n`. In this case, the filter expression is for negation and it is separated by two underscores. Below are the lookup expressions that are supported across different field types.

### Nullable Fields

> The `isnull` lookup expression is automatically added to any of the below field types for all core Nautobot fields and any Nautobot app FilterSet's that inherit from `nautobot.apps.filters.BaseFilterSetBaseFilterSet`.

Reviewer 2: ```suggestion
The `isnull` lookup expression is automatically added to any of the below field types for all core Nautobot fields and any Nautobot app FilterSet's that inherit from `nautobot.apps.filters.BaseFilterSet`.
```
